src/infra/sslQueryRepo.go

```diff
@@ -0,0 +1,223 @@
+package infra
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+
+	"github.com/speedianet/sam/src/domain/entity"
+	"github.com/speedianet/sam/src/domain/valueObject"
+	infraHelper "github.com/speedianet/sam/src/infra/helper"
+)
+
+type SslQueryRepo struct {
+	olsHttpdConfigPath string
+}
+
+type HttpdVhostConfig struct {
+	VirtualHost string
+	FilePath    string
+}
+
+type VhostConfig struct {
+	FilePath    string
+	FileContent string
+}
+
+func NewSslQueryRepo() *SslQueryRepo {
+	return &SslQueryRepo{
+		olsHttpdConfigPath: "/usr/local/lsws/conf/httpd_config.conf",
+	}
+}
+
+func (repo SslQueryRepo) splitSslCertificate(
+	sslCertContent string,
+) ([]entity.SslCertificate, error) {
+	certificates := []entity.SslCertificate{}
+
+	sslCertContentSlice := strings.SplitAfter(sslCertContent, "-----END CERTIFICATE-----\n")
+	for _, sslCertContentStr := range sslCertContentSlice {
+		certificate, err := entity.NewSslCertificate(sslCertContentStr)
+		if err != nil {
+			return certificates, errors.New("SslCertificateError")
+		}
+		certificates = append(certificates, certificate)
+	}
+
+	return certificates, nil
+}
+
+func (repo SslQueryRepo) SslFactory(
+	sslHostname string,
+	sslPrivateKey string,
+	sslCertContent string,
+) (entity.SslPair, error) {
+	var ssl entity.SslPair
+
+	hostname, err := valueObject.NewFqdn(sslHostname)
+	if err != nil {
+		return ssl, errors.New("SslHostnameError")
+	}
+	_, err = repo.GetVhostConfig(hostname.String())
+	if err != nil {
+		return ssl, err
+	}
+
+	privateKey, err := entity.NewSslPrivateKey(sslPrivateKey)
+	if err != nil {
+		return ssl, err
+	}
+
+	sslCertificates, err := repo.splitSslCertificate(sslCertContent)
+	if err != nil {
+		return ssl, err
+	}
+
+	var certificate entity.SslCertificate
+	var chainCertificates []entity.SslCertificate
+	for _, sslCertificate := range sslCertificates {
+		if sslCertificate.IsCA {
+			certificate = sslCertificate
+			continue
+		}
+
+		chainCertificates = append(chainCertificates, sslCertificate)
+	}
```

Comment:
Consegue me explicar o pseudo-código aqui por favor?

Reply:
_Split_ dos certificados em uma _slice_
Declaro a variável `certificate` (certificado **SSL**) 
Declaro a variável `chainCertificates` (certificados intermediários e _root_)
Laço de repetição iterando sobre a _slice_ de certificados:
- Se o certificado da iteração atual for o certificado **SSL** ([IsCA()](https://pkg.go.dev/crypto/x509#Certificate.IsCA)), armazenar em `certificate` e continua o laço
- Se o certificado da iteração atual não for o certificado **SSL**, armazenar no _slice_ `chainCertificates`

Reply:
A propriedade `IsCA` é o retorno do método `IsCA()` da biblioteca `x509`, ela diz se ele é o certificado SSL do site ou não. Todos os outros certificados intermediários e _root_ tem o retorno `IsCA()` como `false`. Eu posso mudar o nome da propriedade para facilitar o entendimento, mas não sei que nome eu colocaria.